### PR TITLE
AzureTRE_Overview

### LINKED ISSUE #2712: AzureTRE Overview Docs Improvement
## Description

As a stakeholder interested in AzureTRE,
I want to have a clear description of the concept of AzureTRE
So that I could apply it to various business scearios

## Acceptance criteria

Update AzureTRE Overview page in AzureTRE documentation: https://microsoft.github.io/AzureTRE/

### PR DESCRIPTION
**Pull request by anastasiakuzn**
# Resolves #2712 

## AzureTRE Overview Modification

This PR adds more elaborate content on the concept of AzureTRE in public facing documentation

## How is this addressed

- Describe the concept in more details with references to other sources
- Replace existing diagram with a more recent version

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
+Across the health industry, be it a pharmaceutical company interrogating clinical trial results, or a public health provider analyzing electronic health records, there is the need to enable researchers, analysts, and developers to work with sensitive data sets.  
+
+Trusted Research Environments (TREs) enable organisations to provide research teams secure access to these data sets along side tooling to ensure a researchers can be productive. Further information on TREs in general can be found in many places, one good resource is [HDR UK](https://www.hdruk.ac.uk/access-to-health-data/trusted-research-environments/).
+
+The Azure Trusted Research Environment project is an accelerator to assist Microsoft customers and partners who want to build out Trusted Research environments on Azure. This project enables authorized users to deploy and configure secure workspaces and researcher tooling without a dependency on IT teams.  
+
 Trusted Research Environments (TRE) enforce a secure boundary around distinct workspaces to enable information governance controls to be enforced.
 
-![Concepts](assets/treconcepts.png)
+![Concepts](assets/treconcepts_updated.png)
```
Comment (marrobi) [inline docs/index.md:13]: Can concepts (here and below) be moved to the hidden terms/glossary piece? I'm wondering if this diagram is better somewhere else and use the prettier blue one here?

Ideally we need some TRE screenshots.

Comment (anastasiakuzn) [reply docs/index.md:13]: @marrobi , okay, would you like me to do the following then: 
1. Move the "concepts" diagram and all the concepts below to a separate sub-section under "Overview" section. This sub-section could be called "Terms and Definitions"
2. Update the diagram to the "blue" one in the "AzureTRE  Overview" page

Comment (marrobi) [reply docs/index.md:13]: That would be great :)

Comment (anastasiakuzn) [reply docs/index.md:13]: alright, I will do that!

**Review by marrobi — APPROVED** [review]
LGTM. Reviewed on mobile so might have missed small things, but think things are in the right place now. Thanks.

**Pull request merged**